application/frontend/src/pages/releases/detail/access-box/gcp-storage-iam-share-form.tsx

```diff
@@ -46,6 +46,9 @@ export const GcpStorageIamShareForm: React.FC<Props> = ({ releaseKey }) => {
 
   const iamUsersArray = iamUsers.split(/[ ,]+/).filter((u) => u);
 
+  const isSuccessState = (status: any): status is SuccessState => true;
+  const isErrorState = (status: any): status is ErrorState => true;
```

Comment:
I like the type guards! Making them more type-safe is only marginally more work:

```suggestion
  const isSuccessState = (status: any): status is SuccessState => 'recordsUpdated' in status;
  const isErrorState = (status: any): status is ErrorState => 'error' in status;
```

(Also, I think the "Something went wrong fetching release data" error will always be displayed even when there is no error unless you make them type-safe).

Reply:
Thanks! Yes - was just fixing the compile errors to get it going - we will need to refactor this form anyhow (we have to work out which bits of functionality move into the control panel and which stays)